### ISSUE TITLE
Changing minInstanceSize not scaling the base instanceSize

### ISSUE THREAD

**Issue by sunchill06** [mongodb/mongodb-atlas-kubernetes#777]
**What did you do to encounter the bug?**

All the following steps are performed on operator release 1.4.1

Created an `AtlasDeployment` using following Spec.
```shell
spec:
  projectRef:
    name: demo-project
  advancedDeploymentSpec:
    backupEnabled: true
    clusterType: REPLICASET
    mongoDBMajorVersion: "4.4"
    name: advanced-cluster
    pitEnabled: false
    replicationSpecs:
    - regionConfigs:
      - autoScaling:
          compute:
            enabled: true
            maxInstanceSize: M40
            minInstanceSize: M10
            scaleDownEnabled: true
          diskGB:
            enabled: true
        electableSpecs:
          instanceSize: M10
          nodeCount: 3
        priority: 7
        providerName: GCP
        regionName: WESTERN_EUROPE
```

Updated the `minInstanceSize` to M20

```shell
          compute:
            enabled: true
            maxInstanceSize: M40
            minInstanceSize: M20
            scaleDownEnabled: true
```
**What did you expect?**
instanceSize (which is currently M10), should be autoscaled to M20 (min size of autoscaling)

This expectation was also inline with this feedback [request](https://feedback.mongodb.com/forums/924355-ops-tools/suggestions/45692755-mongodb-atlas-kubernetes-operator-need-to-adjust-i).

**What happened instead?**

Got this error first:
```console
{"level":"INFO","time":"2022-11-23T13:35:04.409Z","msg":"Status update","atlasdeployment":"platform/advanced-cluster","lastCondition":{"type":"DeploymentReady","status":"False","lastTransitionTime":null,"reason":"DeploymentNotUpdatedInAtlas","message":"PATCH https://cloud.mongodb.com/api/atlas/v1.5/groups/63777688ed29ab0c1b643abf/clusters/advanced-cluster: 400 (request \"BASE_INSTANCE_SIZE_MUST_MATCH\") Electable and read-only nodes must all have the same instance size."}}
```
I expect that we don't need to pass `readOnlySpecs` as they are [optional](https://www.mongodb.com/docs/atlas/reference/atlas-operator/atlasdeployment-custom-resource/#mongodb-setting-spec.advancedDeploymentSpec.replicationSpecs.regionConfigs.readOnlySpecs) but wonder why the operator needs them.

Updated the Spec to include `readOnlySpecs` as well and then getting this error:

```shell
spec:
  advancedDeploymentSpec:
    backupEnabled: true
    clusterType: REPLICASET
    mongoDBMajorVersion: "4.4"
    name: advanced-cluster
    pitEnabled: false
    replicationSpecs:
    - regionConfigs:
      - autoScaling:
          compute:
            enabled: true
            maxInstanceSize: M40
            minInstanceSize: M10
            scaleDownEnabled: true
          diskGB:
            enabled: true
        electableSpecs:
          instanceSize: M10
          nodeCount: 3
        priority: 7
        providerName: GCP
        readOnlySpecs:
          instanceSize: M10
          nodeCount: 3
        regionName: WESTERN_EUROPE
```
New Error:

```console
{"level":"WARN","time":"2022-11-23T13:37:09.424Z","msg":"The instance size is below the minimum autoscaling configuration. Setting it to M20. Consider update your CRD","atlasdeployment":"platform/advanced-cluster"}
{"level":"WARN","time":"2022-11-23T13:37:09.424Z","msg":"The instance size is below the minimum autoscaling configuration. Setting it to M20. Consider update your CRD","atlasdeployment":"platform/advanced-cluster"}
{"level":"INFO","time":"2022-11-23T13:37:09.656Z","msg":"Status update","atlasdeployment":"platform/advanced-cluster","lastCondition":{"type":"DeploymentReady","status":"False","lastTransitionTime":null,"reason":"DeploymentNotUpdatedInAtlas","message":"PATCH https://cloud.mongodb.com/api/atlas/v1.5/groups/63777688ed29ab0c1b643abf/clusters/advanced-cluster: 500 (request \"UNEXPECTED_ERROR\") Unexpected error."}}
```

I also tried to replicate the same scenario by adding `readOnlySpecs` at the first step as part of creating the `AtlasDeployment` but still get the same issue.

**Operator Information**
 - Operator Version - 1.4.1

**Comment by igor-karpukhin** [comment]
Hi @sunchill06, thanks a lot for raising an issue. We will make sure it is fixed in the next release.

**Comment by fabritsius** [comment]
Hey @sunchill06, 

This should be fixed in [v1.5.0](https://github.com/mongodb/mongodb-atlas-kubernetes/releases/tag/v1.5.0). Thanks again for raising this 😌
